test/test_regex_signature.py

```diff
@@ -68,3 +68,7 @@ def test_invalid(self):
         signature = RegexSignature('random', 'Random', '')
         self.assertFalse(signature.match('', ''))
 
+    def test_invalid_regex(self):
+        with self.assertRaises(TypeError):
+            RegexSignature('contents', 'Facebook Secret Key',
+                           '(?i)(facebook|fb)(.{0,20})?(?-i)[''\"][0-9a-f]{32}[''\"]')
```

Comment:
There was 3 rules that had the `(?-i)` rule which raised an error - do you know what this does? I just removed it...

Reply:
Seems to be golang regexp flags. The `(?i)` enables case-insensitive matching for the `(facebook|fb)(.{0,20})?` part and the `(?-i)` disables it again for the `[''\"][0-9a-f]{32}[''\"]` part.

I think the python equivalent would be something lik `(?i)` for enabling case-insensitive matching and `(?i-)` for disabling it again.

Reply:
Alas not :(

```
TypeError: Failed to compile regex for Facebook Secret Key `(?i)(facebook|fb)(.{0,20})?(?1-)['\"][0-9a-f]{32}['\"]` - unknown extension ?1 at position 28
```

Reply:
Looks like the TypeError is caused by a typo `(?1-)` instead of `(?i-)`.
But you're right, there are two ways to specify inline modifiers in python and the `(?i)` one applies globally to the whole regex.
To achieve the desired result of applying the inline modifier only to a specific group of the regex, the grouped way (`(?i:...)` of specifying an inline modifier needs to be used according to the [re module docs](https://docs.python.org/3.8/library/re.html#index-16).

eg. something like:
```
'(?i:(facebook|fb)(.{0,20})?)[''\"][0-9a-f]{32}[''\"]'
```

Reply:
Is that valid? When running with https://regex101.com/ in Python mode it fails.

Is it possible to specify case insensitive for partial.

Reply:
It's only available in python 3.6+, but there it seems to work fine, also with partial case insensitive:
```
Python 3.8.5 (default, Jul 21 2020, 10:42:08) 
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import re
>>> m = re.match('(?i:(facebook|fb)(.{0,20})?)[''\"][0-9a-f]{32}[''\"]', 'faceBOOK="aaaabbbbccccddddaaaabbbbccccdddd"')
>>> print(m)
<re.Match object; span=(0, 43), match='faceBOOK="aaaabbbbccccddddaaaabbbbccccdddd"'>
>>> m = re.match('(?i:(facebook|fb)(.{0,20})?)[''\"][0-9a-f]{32}[''\"]', 'faceBOOK="aaaabbbbccccddddaaaabbbbccccdddD"')
>>> print(m)
None
```

Reply:
Thank for the heads up! I updated the rules and pushed the min-package version to 3.6. @valayDave what do you think?